sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -262,17 +262,20 @@ def _create_pipeline(self, credential, **kwargs):
         return config, Pipeline(config.transport, policies=policies)
 
     def _batch_send(
-        self, *reqs,  # type: HttpRequest
+        self,
+        *reqs,  # type: HttpRequest
+        container_name,  # type: str
```

Comment:
_batch_send will not work if we add delete_blobs on blob_service_client because there won't be a container name there

Reply:
This is only intended for container level batch calls. Once we start having discussions about service level I feel we will probably change the current flow anyway (and maybe even remove or change `batch_send` greatly) and we might utilize generated code. However if you want to avoid passing container name here I can put another workaround. Let me know

Reply:
it's in base_client, that means the change should be in every package, we should make it more generic.
Also we should serialize the query parameter instead of hardcode it. Then less work will be there if we want to add batch on blob_service_client.
If we make it specific for container, higher risk to break container.delete_blobs(), since you need to touch two apis to make them both work.
We can avoid using batch_send if the generated batch_submit() works. It's your call

Reply:
offline: will proceed to make it more generic and later on we will utilize the `batch_submit` operation